cmd_actuators_mux/src/cmd_actuator_mux_nodelet.cpp

```diff
@@ -0,0 +1,436 @@
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of QinetiQ nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+#include "cmd_actuators_mux/cmd_actuator_mux_nodelet.h"
+
+#include <pluginlib/class_list_macros.h>
+#include <std_msgs/String.h>
+
+#include <fstream>
+#include <string>
+
+#include "cmd_actuators_mux/exceptions.h"
+
+namespace cmd_actuator_mux
+{
+/*****************************************************************************
+ ** Implementation
+ *****************************************************************************/
+
+void CmdActuatorMuxNodelet::cmdFinAngleCallback(const fin_control::SetAngles::ConstPtr& msg,
+                                                unsigned int idx)
+{
+  // Reset general timer
+  fin_angles_common_timer.stop();
+  fin_angles_common_timer.start();
+
+  if (cmdActuatorCallbackProcess(cmd_fin_angles_subs, idx))
+  {
+    // The fin angle input has changed from an input to another
+    fin_angles_acv_msg.data = cmd_fin_angles_subs[idx]->name;
+
+    // Wait until both RPM and Fin commands come from the same source
+    if (set_rpm_acv_msg.data == fin_angles_acv_msg.data)
+    {
+      if (publish_new_subscriber)
+      {
+        active_subscriber_pub.publish(fin_angles_acv_msg);
+        publish_new_subscriber = false;
+      }
+      fin_angles_output_topic_pub.publish(msg);
+    }
+    else
+      ROS_DEBUG_STREAM("RPM and fin angles values come from different source"
+                       << " - set RPM source: " << cmd_set_rpm_subs[idx]->name
+                       << " - set Fin Angles source: " << cmd_fin_angles_subs[idx]->name);
```

Comment:
@patapufate consider
```suggestion
                       << " - set RPM source: " << set_rpm_acv_msg.data
                       << " - set Fin Angles source: " << fin_angles_acv_msg.data);
```
which is probably more accurate.

Reply:
Fixed in 5cde148